api/server/handler.go

```diff
@@ -0,0 +1,134 @@
+			return
+		}
+	}
+	defer r.Body.Close()
+	desc, err := m.store.Put(r.Context(), r.Body)
+	switch err {
+	case nil:
+	case blob.ErrBlobTooLarge:
+		message := fmt.Sprintf(`Blob length exceeds the maximum accepted length of %d bytes.`, m.maxBlobLength)
+		respondWithJson(w, api.ErrorResponse{Error: message}, http.StatusBadRequest)
+		return
+	default:
+		respondWithJson(w, api.ErrorResponse{Error: "Internal error occurred while uploading data: " + err.Error()}, http.StatusInternalServerError)
+		return
+	}
+	logger := logger.With("id", desc.ID, "size", desc.Size)
+	if contentLength != 0 && desc.Size != contentLength {
+		logger.Warnw("Content-Length in request header did not match the data length", "expectedSize", contentLength)
+		// TODO: add option to reject such requests?
+	}
+	respondWithJson(w, api.PostBlobResponse{ID: desc.ID.String()}, http.StatusCreated)
+	logger.Debugw("Blob crated successfully", "id", desc.ID, "size", desc.Size)
+}
+
+func (m *HttpServer) handleBlobSubtree(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case http.MethodOptions:
+		w.Header().Set(httpHeaderAllow(http.MethodGet, http.MethodOptions))
+	case http.MethodGet:
+		m.handleBlobGet(w, r)
+	default:
+		respondWithNotAllowed(w, http.MethodPost, http.MethodOptions)
+	}
+}
+
+func (m *HttpServer) handleBlobGet(w http.ResponseWriter, r *http.Request) {
+	suffix := strings.TrimPrefix(r.URL.Path, "/v0/blob/")
+	segments := strings.Split(suffix, "/")
+	switch len(segments) {
+	case 1:
+		m.handleBlobGetByID(w, r, segments[0])
+	case 2:
+		if segments[1] == "status" {
+			m.handleBlobGetStatusByID(w, r, segments[0])
+		} else {
+			respondWithJson(w, api.ErrorResponse{Error: "404 Page Not found"}, http.StatusNotFound)
+		}
+	default:
+		respondWithJson(w, api.ErrorResponse{Error: "404 Page Not found"}, http.StatusNotFound)
+	}
+}
+
+func (m *HttpServer) handleBlobGetByID(w http.ResponseWriter, r *http.Request, idUriSegment string) {
+	var id blob.ID
+	if err := id.Decode(idUriSegment); err != nil {
+		respondWithJson(w, api.ErrorResponse{Error: "Invalid blob ID"}, http.StatusBadRequest)
+		return
+	}
+	logger := logger.With("id", id)
+	blobReader, err := m.store.Get(r.Context(), id)
+	switch err {
+	case nil:
+	case blob.ErrBlobNotFound:
+		respondWithJson(w, api.ErrorResponse{Error: "No blob is found for the given ID"}, http.StatusNotFound)
+		return
+	default:
+		respondWithJson(w, api.ErrorResponse{Error: "Internal error occurred while getting blob: " + err.Error()}, http.StatusInternalServerError)
+		return
+	}
+	w.Header().Set(httpHeaderContentTypeOctetStream())
+	w.Header().Set(httpHeaderContentTypeOptionsNoSniff())
+	w.Header().Set("Content-Disposition", fmt.Sprintf(`attachement; filename="%s.bin"`, id.String()))
+
+	// TODO: Use pooled buffers with configurable size for better efficiency
+	var buf []byte
+	if written, err := io.CopyBuffer(w, blobReader, buf); err != nil {
```

Comment:
consider using https://pkg.go.dev/net/http#ServeContent to also support range requests?

Reply:
Are we confident that we can always get a `ReadSeeker` for a blob?

Reply:
Created this discussion to clarify if we need to support this from get go: https://github.com/filecoin-project/motion/discussions/14

Reply:
Resolved in [e1a7dea](https://github.com/filecoin-project/motion/pull/12/commits/e1a7dead14a5225727f8fd5d3bc226b77ea10415)